lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -734,6 +734,19 @@ static bool isNSObjectHashValue(ValueDecl *baseDecl) {
   return false;
 }
 
+/// Returns true if the given declaration is for the `NSObject.hash(into:)`
+/// function.
+static bool isNSObjectHashMethod(ValueDecl *baseDecl) {
+  ASTContext &ctx = baseDecl->getASTContext();
+
+  if (auto baseVar = dyn_cast<FuncDecl>(baseDecl)) {
```

Comment:
Nit: this name should be `baseFunc` ? 
Also the ctx variable is only used in case is this is true
So this could be 
```cpp
static bool isNSObjectHashMethod(ValueDecl *baseDecl) {
   auto baseFunc = dyn_cast<FuncDecl>(baseDecl);
   if (!baseFunc)
      return false;

   if (auto classDecl = baseFunc->getDeclContext()->getSelfClassDecl()) {
     ASTContext &ctx = baseFunc->getASTContext();
     return baseFunc->getBaseName() == ctx.Id_hash && classDecl->isNSObject();
   }
   return false;
 }
```
WDYT?

Reply:
Fixed! :)

Reply:
Thanks!